src/Acuminator/Acuminator.Analyzers/StaticAnalysis/RowChangesInEventHandlers/EventArgsRowWalker.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Acuminator.Utilities.Roslyn;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Acuminator.Analyzers.StaticAnalysis.RowChangesInEventHandlers
+{
+	public partial class RowChangesInEventHandlersAnalyzer
+	{
+		/// <summary>
+		/// Searches for <code>e.Row</code> mentions
+		/// </summary>
+		private class EventArgsRowWalker : CSharpSyntaxWalker
+		{
+			private static readonly string RowPropertyName = "Row";
+
+			private readonly SemanticModel _semanticModel;
+			private readonly PXContext _pxContext;
+
+			public bool Success { get; private set; }
+
+			public EventArgsRowWalker(SemanticModel semanticModel, PXContext pxContext)
+			{
+				_semanticModel = semanticModel;
+				_pxContext = pxContext;
+			}
+
+			public override void Visit(SyntaxNode node)
+			{
+				if (!Success)
+					base.Visit(node);
+			}
+
+			public override void VisitIdentifierName(IdentifierNameSyntax node)
```

Comment:
@dropsonic this search will find only variables directly assigned with the e.Row. The next order assignments like in the example below are not found:

```
public void MyDac_MyProperty_FieldVerifying(PXCache cache, PXFieldVerifyingEventArgs e)
{
   var dac = e.Row as MyDac;
   var copy = cache.CreateCopy(dac );
   copy.Property = "test"; 
   cache.Update(copy);
}
```

Reply:
It is OK and expected.

Reply:
Well, what about such case:
```
public void MyDac_MyProperty_FieldVerifying(PXCache cache, PXFieldVerifyingEventArgs e)
{
   var dac = e.Row as MyDac;
   var copy = dac
   copy.Property = "test"; 
   
}
```

Reply:
@dropsonic I suggest to consider this as corner case and ignore for now but create a task in JIRA